### PR TITLE
Added RHOS requirements to the docker image

### PR DESCRIPTION
**Pull request by pregnor**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | -
| License         | Apache 2.0

### What's in this PR?

Added stuff required to run on RHOS.

### Why?

We want to support RHOS and non-root privileges are required from the containers.

### Additional context

Tested by @hi-im-aren 

### Checklist

- [X] Code meets the [Developer Guide](https://github.com/banzaicloud/developer-guide)
- ~User guide and development docs updated (if needed)~
- ~Related Helm chart(s) updated (if needed)~

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -38,5 +39,9 @@ RUN mkdir ${KAFKA_HOME} && apt-get update && apt-get install curl -y && apt-get
 COPY --from=kafka_dist /var/tmp/kafka_$scala_version-$kafka_version ${KAFKA_HOME}
 
 RUN chmod a+x ${KAFKA_HOME}/bin/*.sh
+RUN chmod g+rwX ${KAFKA_HOME}
+
+# RUN adduser ${PROJECT_NAME} --disabled-password
```
Comment (pregnor) [inline Dockerfile]: **Note**: to be investigated and either removed or added back in place of the hard coded user.

File: Dockerfile [hunk 2]
```diff
@@ -38,5 +39,9 @@ RUN mkdir ${KAFKA_HOME} && apt-get update && apt-get install curl -y && apt-get
...
+USER 1337
```
Comment (tiswanso) [inline Dockerfile]: This won't work for istio-proxy sidecar injected Kafka brokers

1337 is the UID that istio-proxy runs as and can't have anything else running with that UID due to iptables rules.

https://istio.io/latest/docs/reference/config/analysis/ist0144/

File: Dockerfile [hunk 3]
```diff
@@ -1,8 +1,9 @@
 FROM alpine:latest AS kafka_dist
```
Comment (tiswanso) [inline Dockerfile]: shouldn't we pin the alpine version? 

and defining an ARG with default so the CI build params can potentially build & validate multiple target versions.

Comment (pregnor) [reply Dockerfile]: Yeah, it is a possibility, I haven't looked to fix everything with everything, just opened a PR for Balázs' changes.

File: Dockerfile (hunk 2 above)
Comment (pregnor) [reply Dockerfile]: Is there any specific ID you recommend?

Comment (tiswanso) [reply Dockerfile]: in other nginx containers we've used 1001 or 1000.  I saw prometheus using 65534 or something like that.  I made this and was trying some things:  https://github.com/tiswanso/docker-kafka/commit/6e6a563836c0785c562b35b9b272c32d55cc661f

Comment (tiswanso) [reply Dockerfile]: note in k8s RHOS we'll need to set `runAsUser: 10001` (or whatever matching UID) and we can use `nonroot-v2` policy

**Review by tiswanso — DISMISSED** [review]
lgtm if it works with koperator changes

File: Dockerfile [hunk 4]
```diff
@@ -1,12 +1,19 @@
-FROM alpine:latest AS kafka_dist
+ARG ALPINE_VERSION=latest
+FROM alpine:${ALPINE_VERSION} AS kafka_dist
```
Comment (sagikazarmark) [inline Dockerfile:2]: Can we pin a version instead and let dependabot update the image?

File: Dockerfile [hunk 5]
```diff
@@ -1,12 +1,19 @@
...
 
 ARG scala_version=2.13
-ARG kafka_version=3.1.0
+ARG kafka_version=3.1.2
 ARG kafka_distro_base_url=https://dlcdn.apache.org/kafka
+ARG PROJECT_NAME=koperator
+ARG UID=29092
```
Comment (sagikazarmark) [inline Dockerfile:8]: Are these really supposed to be ARGs?

**Pull request closed without merging**